src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDbScalerProvider.cs

```diff
@@ -0,0 +1,88 @@
+
+namespace Microsoft.Azure.WebJobs.Extensions.CosmosDB.Trigger
+{
+    internal class CosmosDbScalerProvider : IScaleMonitorProvider, ITargetScalerProvider
+    {
+        private readonly CosmosDBScaleMonitor _scaleMonitor;
+        private readonly CosmosDBTargetScaler _targetScaler;
+
+        public CosmosDbScalerProvider(IServiceProvider serviceProvider, TriggerMetadata triggerMetadata) 
+        {
+            AzureComponentFactory azureComponentFactory = null;
+            if (triggerMetadata.Properties != null && triggerMetadata.Properties.TryGetValue(nameof(AzureComponentFactory), out object value))
+            {
+                azureComponentFactory = value as AzureComponentFactory;
+            }
+            else
+            {
+                azureComponentFactory = serviceProvider.GetService<AzureComponentFactory>();
+            }
+
+            ILoggerFactory loggerFactory = serviceProvider.GetService<ILoggerFactory>();
+            CosmosDbMetadata cosmosDbMetadata = JsonConvert.DeserializeObject<CosmosDbMetadata>(triggerMetadata.Metadata.ToString());
+            cosmosDbMetadata.ResolveProperties(serviceProvider.GetService<INameResolver>());
+            IOptions<CosmosClientOptions> options = serviceProvider.GetService<IOptions<CosmosClientOptions>>();
+            ICosmosDBServiceFactory serviceFactory = serviceProvider.GetService<ICosmosDBServiceFactory>();
+            CosmosClient cosmosClient = serviceFactory.CreateService(cosmosDbMetadata.Connection, options.Value);
+            var monitoredContainer = cosmosClient.GetContainer(cosmosDbMetadata.DatabaseName, cosmosDbMetadata.ContainerName);
+            var leaseContainer = cosmosClient.GetContainer(cosmosDbMetadata.DatabaseName, string.IsNullOrEmpty(cosmosDbMetadata.LeaseContainerName) ? CosmosDBTriggerConstants.DefaultLeaseCollectionName : cosmosDbMetadata.LeaseContainerName);
+            _scaleMonitor = new CosmosDBScaleMonitor(triggerMetadata.FunctionName, loggerFactory.CreateLogger<CosmosDBScaleMonitor>(), monitoredContainer, leaseContainer, cosmosDbMetadata.LeaseContainerPrefix);
+            _targetScaler = new CosmosDBTargetScaler(triggerMetadata.FunctionName, cosmosDbMetadata.MaxItemsPerInvocation, monitoredContainer, leaseContainer, cosmosDbMetadata.LeaseContainerPrefix, loggerFactory.CreateLogger<CosmosDBTargetScaler>());
+        }
+
+        public IScaleMonitor GetMonitor()
+        {
+            return _scaleMonitor;
+        }
+
+        public ITargetScaler GetTargetScaler()
+        {
+            return _targetScaler;
+        }
+
+        internal class CosmosDbMetadata
+        {
+            [JsonProperty]
+            public string Connection { get; set; }
+
+            [JsonProperty]
+            public string DatabaseName { get; set; }
+
+            [JsonProperty]
+            public string ContainerName { get; set; }
+
+            [JsonProperty]
+            public string LeaseContainerName { get; set; }
```

Comment:
You also need the LeaseDatabaseName, users can customize this too. If not present, then the DatabaseName is used